lxd/devlxd_instances.go

```diff
@@ -3,13 +3,78 @@ package main
 import (
 	"net/http"
 
+	"github.com/gorilla/mux"
+
+	"github.com/canonical/lxd/lxd/auth"
 	"github.com/canonical/lxd/lxd/device/filters"
 	"github.com/canonical/lxd/lxd/instance"
+	"github.com/canonical/lxd/lxd/request"
+	"github.com/canonical/lxd/lxd/response"
 	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/entity"
 )
 
 type devLXDDeviceAccessValidator func(device map[string]string) bool
 
+var devLXDInstanceEndpoint = APIEndpoint{
+	Path: "instances/{name}",
+	Get:  APIEndpointAction{Handler: devLXDInstanceGetHandler, AccessHandler: allowDevLXDPermission(entity.TypeInstance, auth.EntitlementCanView, "name")},
+}
+
+// devLXDInstanceGetHandler retrieves instance information for the specified instance name.
+// The returned instance information includes only the devices that are accessible to devLXD
+// and are owned by the devLXD identity making the request.
+func devLXDInstanceGetHandler(d *Daemon, r *http.Request) response.Response {
+	inst, err := getInstanceFromContextAndCheckSecurityFlags(r.Context(), devLXDSecurityKey, devLXDSecurityManagementVolumesKey)
+	if err != nil {
+		return response.DevLXDErrorResponse(err)
+	}
+
+	// Allow access only to the projectName where current instance is running.
+	projectName := inst.Project().Name
+	targetInstName := mux.Vars(r)["name"]
+
+	// Get identity from the request context.
+	identity, err := request.GetCallerIdentityFromContext(r.Context())
+	if identity == nil {
+		return response.DevLXDErrorResponse(err)
+	}
+
+	// Fetch instance.
+	targetInst := api.Instance{}
+
+	url := api.NewURL().Path("1.0", "instances", targetInstName).Project(projectName).WithQuery("recursion", "1").URL
```

Comment:
No need for `.URL` on the end, you can just use `api.NewURL()` returned and call `.String()`  on it when needed.